crates/oxc_ast/src/node.rs

```diff
@@ -0,0 +1,374 @@
+use std::{
+    hash::{Hash, Hasher},
+    ops::Range,
+};
+
+use serde::Serialize;
+
+use crate::{context::Context, Argument, AssignmentTarget, AssignmentTargetPattern, BindingPattern, BindingPatternKind, ClassElement, Declaration, ExportDefaultDeclarationKind, Expression, ForStatementInit, ImportAttributeKey, JSXElementName, MemberExpression, ModuleExportName, ObjectPatternProperty, ObjectProperty, PropertyKey, PropertyValue, SimpleAssignmentTarget, Statement, TSModuleDeclarationName, TSSignature, TSType};
+
+pub type Span = Range<usize>;
+
+#[derive(Debug, Default, Clone, Copy, Serialize, PartialEq, Eq)]
+pub struct Node {
+    pub start: usize,
```

Comment:
What's your reasoning behind using `usize` for node ranges. Isn't a `u32` sufficient for "all" reasonable programs (I never tried executing a 4GB JS file)

Reply:
Good catch! It ought to be `u32` but it was tedious to convert it back to usize in some areas of the code - I think it was mostly the diagnostics printing crate `miette`.